kubernetes/secret.go

```diff
@@ -18,12 +19,23 @@ type RemoteSecretClusterListItem struct {
 }
 
 type RemoteSecretUser struct {
-	Name string                `yaml:"name"`
-	User RemoteSecretUserToken `yaml:"user"`
+	Name string                   `yaml:"name"`
+	User RemoteSecretUserAuthInfo `yaml:"user"`
 }
 
-type RemoteSecretUserToken struct {
-	Token string `yaml:"token"`
+type RemoteSecretUserAuthInfo struct {
+	Token string                `yaml:"token"`
+	Exec  *RemoteSecretUserExec `yaml:"exec"`
+}
+
+type RemoteSecretUserExec struct {
+	Command            string                  `yaml:"command"`
+	Args               []string                `yaml:"args"`
+	Env                []api.ExecEnvVar        `yaml:"env"`
+	APIVersion         string                  `yaml:"apiVersion"`
+	InstallHint        string                  `yaml:"installHint"`
+	ProvideClusterInfo bool                    `yaml:"provideClusterInfo"`
+	InteractiveMode    api.ExecInteractiveMode `yaml:"interactiveMode"`
 }
```

Comment:
Do we need this struct? Can we just use `k8s.io/client-go/tools/clientcmd/api.ExecConfig` directly instead of having to copy over all the struct fields and some defaulting code?

Reply:
I think it's fine to include this as is and we can look at cleaning it up in a later change.